Render whole list using fragment

totokartonio opened the pull request:
1. Изменил способ добавления задачи в DOM, чтобы избавиться от двух источников истины.
2. Добавил chunk CSS Reset.

Commit:
Render whole list using fragment

1. Изменил способ добавления задачи в DOM, чтобы
избавиться от двух источников истины.
2. Добавил chunk CSS Reset.

iffjey commented on script.js:
```diff
@@ -26,43 +26,49 @@ const checkboxHandler = taskId => event => {
+    fragment.append(taskElement);
+  });
+
+  listOfTasks.replaceChildren(fragment);
```
круто, не знал о таком методе!

iffjey commented on script.js:
```diff
@@ -26,43 +26,49 @@ const checkboxHandler = taskId => event => {
+    checkbox.checked = task.completed;
+    checkbox.classList.add('checkbox');
 
-  //Наименование задачи
-  const titleElement = document.createElement('p');
-  titleElement.textContent = task.title;
+    //eventListener для отслеживания состояния checkbox
+    checkbox.addEventListener('change', checkboxHandler(task.id));
 
-  //Кнопка удаления
-  const deleteButton = document.createElement('button');
-  deleteButton.type = 'button';
-  deleteButton.classList.add('delete-button');
+    //Наименование задачи
+    const titleElement = document.createElement('p');
+    titleElement.textContent = task.title;
 
-  //eventListener для удаления задачи
-  deleteButton.addEventListener('click', deleteTaskHandler(task.id, taskElement));
+    //Кнопка удаления
+    const deleteButton = document.createElement('button');
+    deleteButton.type = 'button';
+    deleteButton.classList.add('delete-button');
 
-  taskElement.append(checkbox, titleElement, deleteButton);
+    //eventListener для удаления задачи
+    deleteButton.addEventListener('click', deleteTaskHandler(task.id, taskElement));
 
-  listOfTasks.append(taskElement);
+    taskElement.append(checkbox, titleElement, deleteButton);
```
единственное я бы вот это вынес в отдельную функцию `renderTask` 

```
const renderTask = (task) => HTMLElement 

// чтобы потом в forEach ты мог бы просто сделать 

const taskElement = renderTask(task);
fragment.append(taskElement)
```

iffjey commented on style.css:
```diff
@@ -1,3 +1,54 @@
+  margin: 0;
+}
+
+body {
+  line-height: 1.5;
+  -webkit-font-smoothing: antialiased;
+}
+
+img,
+picture,
+video,
+canvas,
+svg {
+  display: block;
+  max-width: 100%;
+}
+
+input,
+button,
+textarea,
+select {
+  font: inherit;
+}
+
+p,
+h1,
+h2,
+h3,
+h4,
+h5,
+h6 {
+  overflow-wrap: break-word;
+}
+
+#root,
+#__next {
+  isolation: isolate;
+}
+/* End of CSS Reset */
+
```
ещё ты можешь создать файл `reset.css` и просто подключить его в html, чтобы он не хранился тут

Commit:
Add renderTask function

1. Теперь для рендеринга задач выделена отдельная
функция.
2. Файлы CSS были разделены на reset и style.

Commit:
Corrected small things
